cubic-server/nbt.hpp

```diff
@@ -429,11 +432,45 @@ class List : public Base
                 current = i->getType();
             if (current != i->getType())
                 throw std::runtime_error("nbt::List contains more than one type");
-            i->serialize(data, false);
+            // i->serialize(data, false);
+            nbt::serialize(data, i, false);
         }
     }
 };
 
+constexpr void serialize(std::vector<uint8_t> &data, const Base *nbt, bool include_name)
+{
+    switch (nbt->getType())
+    {
+    case TagType::Byte:
+        return ((const Byte *)nbt)->serialize(data, include_name);
+    case TagType::Short:
+        return ((const Short *)nbt)->serialize(data, include_name);
+    case TagType::Int:
+        return ((const Int *)nbt)->serialize(data, include_name);
+    case TagType::Long:
+        return ((const Long *)nbt)->serialize(data, include_name);
+    case TagType::Float:
+        return ((const Float *)nbt)->serialize(data, include_name);
+    case TagType::Double:
+        return ((const Double *)nbt)->serialize(data, include_name);
+    case TagType::ByteArray:
+        return ((const ByteArray *)nbt)->serialize(data, include_name);
+    case TagType::String:
+        return ((const String *)nbt)->serialize(data, include_name);
+    case TagType::List:
+        return ((const List *)nbt)->serialize(data, include_name);
+    case TagType::Compound:
+        return ((const Compound *)nbt)->serialize(data, include_name);
+    case TagType::IntArray:
+        return ((const IntArray *)nbt)->serialize(data, include_name);
+    case TagType::LongArray:
+        return ((const LongArray *)nbt)->serialize(data, include_name);
+    default:
```

Comment:
?????

Reply:
So ... actually, there is a problem. When creating the packet with the registery-codec (the big nbt), there is a segfault. This was an attempt to try to fix it, but it's not working at all

Reply:
![image](https://user-images.githubusercontent.com/60519602/209223185-db36b2a7-5048-43e9-abf8-662bb11443f2.png)
![image](https://user-images.githubusercontent.com/60519602/209223271-566bdad0-f3ed-496d-aed2-763ae0a62aa2.png)
![image](https://user-images.githubusercontent.com/60519602/209223334-cfb782d5-18cc-4a79-b806-f44165c8f1e5.png)
as you can see there, some values are weird and some other are good. Just after the call to serialize, the program segfault.

Reply:
I will check that out tonight, I'll update you on this.